internal/app/replication.go

```diff
@@ -57,12 +58,27 @@ func (app *App) TryRepairReplication(node *mysql.Node, master string) {
 	replState.LastAttempt = time.Now()
 }
 
-func StartSlaveAlgorithm(app *App, node *mysql.Node, _ string) error {
+func (app *App) makeReplStateKey(node *mysql.Node, channel string) string {
+	if channel == app.config.ExternalReplicationChannel {
+		return fmt.Sprintf("%s-%s", node.Host(), channel)
+	}
+	return node.Host()
+}
+
+func StartSlaveAlgorithm(app *App, node *mysql.Node, _ string, channel string) error {
 	app.logger.Infof("repair: trying to repair replication using StartSlaveAlgorithm...")
+	if channel == app.config.ExternalReplicationChannel {
+		return node.StartExternalReplication()
+	}
 	return node.StartSlave()
 }
 
-func ResetSlaveAlgorithm(app *App, node *mysql.Node, master string) error {
+func ResetSlaveAlgorithm(app *App, node *mysql.Node, master string, channel string) error {
+	// TODO we don't want reset slave on external replication
```

Comment:
remove TODO?

Reply:
No, at least not this time...
May be split algorithms in external/internal later.
Will write this in TODO